packages/libsodium.rb

```diff
@@ -3,35 +3,53 @@
 class Libsodium < Package
   description 'A modern, portable, easy to use crypto library'
   homepage 'https://libsodium.org'
-  version '1.0.18'
+  version '1.0.18-7168'
   license 'ISC'
   compatibility 'all'
-  source_url 'https://download.libsodium.org/libsodium/releases/libsodium-1.0.18.tar.gz'
-  source_sha256 '6f504490b342a4f8a4c4a02fc9b866cbef8622d5df4e5452b46be121e46636c1'
+  source_url 'file:///dev/null'
+  source_sha256 'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855'
 
-  binary_url ({
-    aarch64: 'https://downloads.sourceforge.net/project/chromebrew/armv7l/libsodium-1.0.18-chromeos-armv7l.tar.xz',
-     armv7l: 'https://downloads.sourceforge.net/project/chromebrew/armv7l/libsodium-1.0.18-chromeos-armv7l.tar.xz',
-       i686: 'https://downloads.sourceforge.net/project/chromebrew/i686/libsodium-1.0.18-chromeos-i686.tar.xz',
-     x86_64: 'https://downloads.sourceforge.net/project/chromebrew/x86_64/libsodium-1.0.18-chromeos-x86_64.tar.xz',
+  binary_url({
+    aarch64: 'https://downloads.sourceforge.net/project/chromebrew/armv7l/libsodium-1.0.18-7168-chromeos-armv7l.tar.xz',
+     armv7l: 'https://downloads.sourceforge.net/project/chromebrew/armv7l/libsodium-1.0.18-7168-chromeos-armv7l.tar.xz',
+       i686: 'https://downloads.sourceforge.net/project/chromebrew/i686/libsodium-1.0.18-7168-chromeos-i686.tar.xz',
+     x86_64: 'https://downloads.sourceforge.net/project/chromebrew/x86_64/libsodium-1.0.18-7168-chromeos-x86_64.tar.xz'
   })
-  binary_sha256 ({
-    aarch64: 'fa8d8b5496d8d1ae4e98a7c8f118898efdcb2c4b92f235ea12f73b12b36ff80e',
-     armv7l: 'fa8d8b5496d8d1ae4e98a7c8f118898efdcb2c4b92f235ea12f73b12b36ff80e',
-       i686: '5b2161327ebf6573fc0660bea7ef75702cd593bf3c0da0cab11a007a89567fbb',
-     x86_64: 'a2dfec82f2c2c4feba5b6c3b64e61977e5f69d4ea2279266443c69ebfcddc3a7',
+  binary_sha256({
+    aarch64: '8cf24abf23f1ec6b19052e29b5cc80db8f2b6917dae193827c99a02349839221',
+     armv7l: '8cf24abf23f1ec6b19052e29b5cc80db8f2b6917dae193827c99a02349839221',
+       i686: 'ab1a37ec01e04271deca45cfb8943dc8d379e12c4d154a4d190824e1de6c0daa',
+     x86_64: 'f5ad9b26a80254a30e8aaced54eeb19e8962980b888df9d904a6377d7dce9cff'
   })
 
   def self.build
-    system "./configure", "--prefix=#{CREW_PREFIX}", "--libdir=#{CREW_LIB_PREFIX}"
-    system "make"
+    @git_dir = 'libsodium_git'
+    @git_hash = '7168944f16e12c1b630e66a7b9be0802d5b564dd'
+    @git_url = 'https://github.com/jedisct1/libsodium.git'
+    FileUtils.rm_rf(@git_dir)
+    FileUtils.mkdir_p(@git_dir)
+    Dir.chdir @git_dir do
+      system 'git init'
+      system "git remote add origin #{@git_url}"
+      system "git fetch --depth 1 origin #{@git_hash}"
+      system 'git checkout FETCH_HEAD'
+      system 'NOCONFIGURE=1 ./autogen.sh'
+      system "env CFLAGS='-flto=auto' CXXFLAGS='-flto=auto' \
+      LDFLAGS='-flto=auto'\
+      ./configure #{CREW_OPTIONS}"
+      system 'make'
+    end
```

Comment:
Could this take advantage of merging https://github.com/skycocker/chromebrew/pull/5598?

Reply:
This is **exactly** what #5998 should be able to handle.

Reply:
@satmandu you might not know about these lines of code in lib/const.rb
```
CREW_COMMON_FLAGS = "'-Os -pipe -flto=auto -fuse-ld=gold'"
CREW_COMMON_FNO_LTO_FLAGS = "'-Os -pipe -fno-lto -fuse-ld=gold'"
CREW_FNO_LTO_LDFLAGS = "'-fno-lto'"
CREW_LDFLAGS = "'-flto=auto'"

CREW_ENV_OPTIONS = "CFLAGS=#{CREW_COMMON_FLAGS} CXXFLAGS=#{CREW_COMMON_FLAGS} FCFLAGS=#{CREW_COMMON_FLAGS} FFLAGS=#{CREW_COMMON_FLAGS} LDFLAGS=#{CREW_LDFLAGS}"
```

This is an effort to standardize flags across packages.

Reply:
@saltedcoffii Thanks! @uberhacker clued me into these earlier, and I've started using them.